ext/src/ruby_api/store.rs

```diff
@@ -391,9 +468,24 @@ impl ResourceLimiter for TrackingResourceLimiter {
         maximum: Option<usize>,
     ) -> anyhow::Result<bool> {
         let res = self.inner.memory_growing(current, desired, maximum);
-        if res.is_ok() && maximum.map_or(true, |maximum| desired <= maximum) {
+
+        // Update max_linear_memory_consumed
+        self.max_linear_memory_consumed = self.max_linear_memory_consumed.max(desired);
+
+        // Check against our stored memory_size limit
+        if let Some(memory_size_limit) = self.memory_size_limit {
+            if desired > memory_size_limit {
+                self.linear_memory_limit_hit = true;
+                return Ok(false);
+            }
+        }
+
+        if res.is_ok() {
             self.tracker.increase_memory_usage(desired - current);
+        } else {
+            self.linear_memory_limit_hit = true;
```

Comment:
`self.inner.memory_growing` is [already checking](https://docs.rs/wasmtime/latest/src/wasmtime/runtime/limits.rs.html#334-352) the following invariants:

* `desired > limit` (which is the same `memory_size` that we're passing through the limits hash)
* `desired > max (if defined)`


With that in mind, I wonder if we can simplify this method to:

```rust
 // Update max_linear_memory_consumed
 self.max_linear_memory_consumed = self.max_linear_memory_consumed.max(desired);
       
if res.is_ok() {
  self.tracker.increase_memory_usage(desired - current);
} else {
  self.linear_memory_limit_hit = true;
}
res
```